File: m26-cs/M26/Joakimsoftware.M26/src/RunWalkCalculation.cs
using System;

// Instances of this class represent an Age of an athlete.
// Heart-rate training zones are calculated for the age.
// Chris Joakim, 2021/07/19

namespace Joakimsoftware.M26 {

    public struct RunWalkCalculation {  

        // Input fields:
        public string runHHMMSS;
        public string runPPM;
        public string walkHHMMSS;
        public string walkPPM;
        public double miles;

        // Calculated fields:
        public Speed averageSpeed;
        public string projectedTime;
    } 

}

File: m26-js-ts/src/RunWalkCalculation.ts

export class RunWalkCalculation {
    
    // Input fields
    public runHhmmss: string;
    public runPpm: string;
    public walkHhmmss: string;
    public walkPpm: string;
    public miles: number;

    // Calculated fields
    public avgMph: number;
    public avgPpm: string;
    public projectedTime:  string;
    public projectedMiles: number;

    constructor() {
    }
}
